### PR TITLE
MISP-1736 preference file not found

### PR DESCRIPTION
**Pull request by CoenSchalkwijk**
Commits are clean, have a look at them separately for easy reviewing :)

### CODE REVIEW

File: libSmeagol/nonVolatilePocket.py [hunk 1]
```diff
@@ -42,40 +43,40 @@ def forceSave(self) -> None:
 
     ## Stops the registry and erases all settings (remove file).
     #  @param restart_after_erase If true then the registryfile will be reopened and the thread started again
-    def erase(self, *, restart_after_erase: bool=True) -> None:
+    def erase(self, *, restart_after_erase: bool = True) -> None:
         self.stop()
         try:
-            log.info("Wiping all registry keys...")
+            log.info("Wiping all registry keys (%s)", self.__preferences_file)
```
Comment (shanx) [inline libSmeagol/nonVolatilePocket.py:49]:
```suggestion
            log.info(f"Wiping all registry keys ({self.__preferences_file})")
```

File: libSmeagol/nonVolatilePocket.py [hunk 2]
```diff
@@ -42,40 +43,40 @@ def forceSave(self) -> None:
...
             os.remove(self.__preferences_file)
-        except OSError as exception:
-            log.error("Unable to remove settings file '%s', because %s", self.__preferences_file, exception.strerror)
-            pass
+        except OSError:
+            log.exception("Unable to remove settings file (%s)", self.__preferences_file)
```
Comment (shanx) [inline libSmeagol/nonVolatilePocket.py:52]:
```suggestion
            log.exception(f"Unable to remove settings file: {self.__preferences_file}")
```

File: libSmeagol/nonVolatilePocket.py [hunk 3]
```diff
@@ -42,40 +43,40 @@ def forceSave(self) -> None:
...
-    ## Implements a special function that will erase all settings, except the keys mentioned in the keys_to_save list and add the extra settings at the end
+    ## Implements a special function that will erase all settings,
+    # except the keys mentioned in the keys_to_save list and add the extra settings at the end
     #  @param keys_to_backup The list of keys to backup prior to removing all settings
-    #  @param settings_to_add The dictionary of key/values to add after te reset and restoration of the backed up key/values
+    #  @param settings_to_add The dictionary to add after a reset and restore of the backed-up key/values
     def backupAndSetup(self, keys_to_backup: List[str], settings_to_add: Dict[str, Any]) -> None:
         saved = {}
-        log.info("Backing up keys...")
+        log.info("Backing up keys (%s)", self.__preferences_file)
```
Comment (shanx) [inline libSmeagol/nonVolatilePocket.py:65]:
```suggestion
        log.info(f"Backing up keys ({self.__preferences_file})")
```

File: libSmeagol/nonVolatilePocket.py [hunk 4]
```diff
@@ -42,40 +43,40 @@ def forceSave(self) -> None:
...
         for key in keys_to_backup:
             saved[key] = self.get(key)
 
         self.erase()
 
-        log.info("Restoring settings...")
+        log.info("Restoring settings (%s)", self.__preferences_file)
```
Comment (shanx) [inline libSmeagol/nonVolatilePocket.py:71]:
```suggestion
        log.info(f"Restoring settings ({self.__preferences_file})")
```

File: libSmeagol/nonVolatilePocket.py [hunk 5]
```diff
@@ -42,40 +43,40 @@ def forceSave(self) -> None:
...
         for key, value in saved.items():
             self.set(key, value)
 
-        log.info("Adding extra settings...")
+        log.info("Adding extra settings (%s)", self.__preferences_file)
```
Comment (shanx) [inline libSmeagol/nonVolatilePocket.py:75]:
```suggestion
        log.info(f"Adding extra settings ({self.__preferences_file})")
```

File: libSmeagol/nonVolatilePocket.py [hunk 6]
```diff
@@ -85,35 +86,29 @@ def _executeAction(self) -> None:
...
-
-    ## Default location for file storage
-    __BASE_PATH = "/var/lib/griffin/"
-
     ## Loads the settings from the specified file in json format
     def __load(self) -> None:
         preferences = {}  # type: Dict[str, Any]
         if os.path.isfile(self.__preferences_file):
-            log.info("Reading preferences from " + self.__preferences_file)
+            log.info("Reading preferences (%s)", self.__preferences_file)
```
Comment (shanx) [inline libSmeagol/nonVolatilePocket.py:100]:
```suggestion
            log.info(f"Reading preferences ({self.__preferences_file})")
```

File: libSmeagol/nonVolatilePocket.py [hunk 7]
```diff
@@ -85,35 +86,29 @@ def _executeAction(self) -> None:
...
             try:
                 with open(self.__preferences_file, "r") as f:
                     preferences = json.load(f)
-            except Exception:
-                log.warning("Error reading preferences file " + self.__preferences_file)
+            except Exception:  # pylint: disable=broad-except
+                log.warning("Error reading preferences file: '%s'", self.__preferences_file)
```
Comment (shanx) [inline libSmeagol/nonVolatilePocket.py:105]:
```suggestion
                log.warning(f"Error reading preferences file: {self.__preferences_file}")
```

File: libSmeagol/nonVolatilePocket.py [hunk 8]
```diff
@@ -122,19 +117,23 @@ def __save(self) -> None:
...
+
             # Open the directory containing the preference file, and fsync it. This forces the rename to disk.
             if hasattr(os, "O_DIRECTORY"):
                 dir_fd = os.open(directory, os.O_DIRECTORY | os.O_RDONLY)
                 os.fsync(dir_fd)
                 os.close(dir_fd)
-        except Exception:
-            log.exception("Error writing preferences " + self.__preferences_file)
+        except Exception:  # pylint: disable=broad-except
+            log.exception("Error writing preferences file '%s'", self.__preferences_file)
```
Comment (shanx) [inline libSmeagol/nonVolatilePocket.py:137]:
```suggestion
            log.exception(f"Error writing preferences file: {self.__preferences_file}")
```

File: libSmeagol/nonVolatilePocket.py [hunk 9]
```diff
@@ -122,19 +117,23 @@ def __save(self) -> None:
...
+
+        os.sync()
```
Comment (shanx) [inline libSmeagol/nonVolatilePocket.py:139]: why the sync here?

File: libSmeagol/pocket.py [hunk 10]
```diff
@@ -103,19 +107,22 @@ def getAsSubPocket(self, key: str, default=None) -> Optional["Pocket"]:
                 sub_pocket.__on_changed.connect(self._handleOnChangeEvent)
                 return sub_pocket
             else:
-                log.warning("Getting key '%s' as pocket for pocket '%s' fails for not being a dictionary but a %s!", key, self, type(value))
+                log.warning(
+                    "Getting key '%s' as pocket for pocket '%s' fails for not being a dictionary but a %s!",
+                    key, self, type(value)
+                )
```
Comment (shanx) [inline libSmeagol/pocket.py:113]:
```suggestion
                log.warning(
                    "Getting key '{key}' as pocket for pocket '{self}' fails for not being a dictionary but: {type(value)}"
                )
```

File: libSmeagol/timerPocket.py [hunk 11]
```diff
@@ -81,7 +81,7 @@ def __callExecuteAction(self) -> None:
     def __startThread(self) -> None:
         self.__running = True
         thread_name = self._getRegistryId()
-        log.info("Setting up thread for " + thread_name)
+        log.info("Setting up thread '%s'", thread_name)
```
Comment (shanx) [inline libSmeagol/timerPocket.py:84]:
```suggestion
        log.info(f"Setting up thread: {thread_name}")
```
What info do we gain from this?

**Review by shanx — COMMENTED** [review]
some first remarks

File: libSmeagol/nonVolatilePocket.py (hunk 1 above)
Comment (CoenSchalkwijk) [reply libSmeagol/nonVolatilePocket.py:49]: For logging, I'd like to stick to lazy formatting.

File: libSmeagol/nonVolatilePocket.py (hunk 9 above)
Comment (robinmdh) [reply libSmeagol/nonVolatilePocket.py:139]: I remember we added that in the opinicus version to improve retention of settings files during test when the machine is power-cycled moments later...

**Review by robinmdh — DISMISSED** [review]

File: libSmeagol/nonVolatilePocket.py (hunk 1 above)
Comment (shanx) [reply libSmeagol/nonVolatilePocket.py:49]: we decided to do that for debug only

**Review by jbatong — DISMISSED** [review]

File: libSmeagol/timerPocket.py (hunk 11 above)
Comment (CoenSchalkwijk) [reply libSmeagol/timerPocket.py:84]: To make it explicit that a thread has started/wil start.
Should a timer event somehow get botched, this at least show the processing thread was started

File: libSmeagol/nonVolatilePocket.py (hunk 9 above)
Comment (CoenSchalkwijk) [reply libSmeagol/nonVolatilePocket.py:139]: What @robinmdh says.
When the user decides to cut the power, settings might not have been written to disc yet.

**Review by jbatong — APPROVED** [review]

**Review by jackha — APPROVED** [review]
lgtm; did not look at unit tests for TimerPocket and maybe out of scope for this issue... testRegistry would sometimes fail mysteriously because it leaves threads running. Tests for TimerPocket might have that problem as well.

**Pull request merged**